deliver/spec/submit_for_review_spec.rb

```diff
@@ -66,11 +71,74 @@ def make_fake_version
         let(:fake_app) { make_fake_app }
         let(:fake_version) { make_fake_version }
         let(:fake_builds) { make_fake_builds(1) }
+
         it 'finds the one build' do
           allow(fake_app).to receive(:latest_version).and_return(fake_version)
           allow(fake_version).to receive(:candidate_builds).and_return(fake_builds)
           only_build = fake_builds.first
-          expect(review_submitter.wait_for_build(fake_app)).to eq(only_build)
+          expect(review_submitter.wait_for_build(fake_app, "1.2.3")).to eq(only_build)
+        end
+      end
+
+      context 'has candidates and one build, but no app version' do
+        let(:fake_app) { make_fake_app }
+        let(:fake_builds) do
+          builds = make_fake_builds(1)
+          builds[0][:train_version] = "1.2.3"
```

Comment:
You should use `double` and `allow`.

```ruby
build = double('fake_build')
allow(build).to receive(:[]).with(:train_version).and_return("1.2.3")
allow(build).to receive(:[]).with(:build_version).and_return("1")
allow(build).to receive(:[]).with(:processing).and_return(false)
[build]
```

Reply:
I don't really know Ruby, so I'm totally willing to defer to your expertise, but would you mind explaining the advantage of stubbing with rspec as opposed to just setting the value? To my untrained eye, it just looks more verbose while accomplishing the same thing.

Reply:
According to RSpec documentation.
https://relishapp.com/rspec/rspec-mocks/docs/basics/test-doubles

`any message you have not allowed or expected will trigger an error`

I think that is the main benefit using Test Doubles.
Using test doubles, we can notice unexpected method callings.